sklearn/tree/_criterion.pyx

```diff
@@ -599,6 +600,95 @@ cdef class Gini(ClassificationCriterion):
         impurity_right[0] = gini_right / self.n_outputs
 
 
+cdef class HellingerDistance(ClassificationCriterion):
+    r"""Hellinger distance criterion.
+
+    This handles cases where the target is a classification taking values 0 and 1.
+    If k represents a node before split, k1 and k2 represent the children nodes after split,
+    left_0 represents observations of class 0 in node k1, right_0 represents observations of class 0 in node k2,
+    left_1 represents observations of class 1 in node k1, right_1 represents observations of class 1 in node k2,
+    parent_0 represents observations of class 0 in node k, parent_1 represents observations of class 1 in node k,
+    then let
```

Comment:
I find list and shorter sentence easier to read about. What do you think?
```suggestion
    
     Let:
       - k represents a node before split
       - k1 and k2 represent the children nodes of k after split
       - left_0 represents observations of class 0 in node k1
       - right_0 represents observations of class 0 in node k2
       - left_1 represents observations of class 1 in node k1
       - right_1 represents observations of class 1 in node k2
       - parent_0 represents observations of class 0 in node k
       - parent_1 represents observations of class 1 in node k
    
    then let
```

Also I would highly recommend:
 - making the mathematical notations of the user guide and the ones here match if possible
 - indicating the support of this criterion i.e. $\left[0, \sqrt{2}\right]$

Reply:
- changed to list
- added a note on hellnger distance score range
- the doc here is pretty similar to the one in tree.rst, with one difference that here I explain the formula bottom up and in tree.rst top down, both cases are clear I think, but please let me know if you want me to change anything else